handlers/cloner.py

```diff
@@ -0,0 +1,46 @@
+import sys
+import subprocess as sp
+from ghooklistener import Listener, PayloadType, HandleFuncReturnType
+
+
+def handlefunc(data: PayloadType) -> HandleFuncReturnType:
+    ref = data["ref"]
+    if ref != "refs/heads/master":
+        print("Not master branch.  Not updating.")
+        return True, "OK"
+
+    # if it was a force push, we will have to fetch and reset
+    forced = data["forced"]
+
+    if pull(forced):
+        return True, "OK"
+
+    return False, "Pull failed"
+
+
+def pull(force: bool) -> bool:
+    if force:
+        print("Force pushes not yet supported")
+        print("Doing normal push")
+    cloneloc = "/tmp/place"
```

Comment:
The clone directory will probably be handled differently in the next update. Maybe a json config file for the service, that contains a mapping of supported git-repositories and the corresponding directories where the git pull is supposed to be performed.

Reply:
The config json file could then also contain the token we use as a secret for verification.